protocol-units/execution/opt-executor/src/executor/mod.rs

```diff
@@ -41,15 +41,16 @@ impl Executor {
 		Arc::clone(&self.db().reader)
 	}
 
-	pub fn decrement_transactions_in_flight(&self, count: u64) {
-		let current = self.transactions_in_flight.get_count();
+	pub async fn decrement_transactions_in_flight(&self, count: u64) {
+		let mut transactions_in_flight = self.transactions_in_flight.write().await;
+		let current = transactions_in_flight.get_count();
 		info!(
 			target: "movement_timing",
 			count,
 			current,
 			"decrementing_transactions_in_flight",
 		);
-		self.transactions_in_flight.decrement(count);
+		transactions_in_flight.decrement(count);
```

Comment:
As I mentioned in Slack, there is nothing that needs to hold the mutex lock across await points. So the mutex could be `std::sync::Mutex`, which is better for performance.

Reply:
Okay, I will introduce this at the risk of someone later modifying, at which point I will link this comment.